src/server/auth/server/api_server.go

```diff
@@ -1866,8 +1977,7 @@ func (a *apiServer) SetACLInTransaction(
 		} else if !strings.HasSuffix(err.Error(), "not found") {
 			// Unclear if repo exists -- return error
 			return false, errors.Wrapf(err, "could not inspect \"%s\"", req.Repo)
-		} else if len(newACL.Entries) == 1 &&
-			newACL.Entries[callerInfo.Subject] == auth.Scope_OWNER {
+		} else if newACL.Entries[callerInfo.Subject] == auth.Scope_OWNER {
```

Comment:
AFAICT this is belt-and-suspenders and it's safe to remove the check that the new ACL length is 1 when creating an ACL for a new repo.

Reply:
Basically yes. The thinking was that ideally only PFS itself would be able to create an ACL for a new repo, and `SetACL` would just require the repo to exist in all cases. This was basically trying to limit what's possible in the weird case that an end-user (possibly malicious) is directly calling `SetACL` on a repo that doesn't exist yet.

Right now, setting an ACL for a nonexistant repo mostly doesn't do anything because CreateRepo calls `SetACL` with an ACL that it constructs from scratch, but this check was a backup for CreateRepo's behavior, basically.

I guess an attack would be possible if someone changed PFS in the future to use `SetScope(<creator>, OWNER)` instead of `SetACL` in CreateRepo. Then you could imagine an attack where:
1. Mallory calls `SetACL(future, {Mallory: OWNER, Alice: OWNER})`, which succeeds b/c there's no repo.
2. Alice calls `CreatePipeline(future) -> CreateRepo(future) -> SetScope(future, Alice, OWNER)`
3. Even though the ACL exists, Alice passes the permission check. Then Mallory can see the pipeline output even though Alice never added them to the ACL, and stays on it basically until someone calls `GetACL` and notices.